cli/src/main/java/de/jplag/cli/CLI.java

```diff
@@ -250,7 +252,17 @@ private String generateDescription() {
         return String.format(DESCRIPTION_PATTERN, randomDescription, CREDITS);
     }
 
-    public String getResultFolder() {
-        return this.options.resultFolder;
+    private String getResultFilePath() {
+        String optionValue = this.options.resultFile;
+        if (optionValue.endsWith(DEFAULT_FILE_ENDING)) {
+            return optionValue;
+        } else {
+            return optionValue + DEFAULT_FILE_ENDING;
+        }
+    }
+
+    private String getResultFileBaseName() {
+        String defaultOutputFile = getResultFilePath();
+        return defaultOutputFile.substring(0, defaultOutputFile.length() - DEFAULT_FILE_ENDING.length());
     }
```

Comment:
Could you add this to the PR description, right now this is a sneaky change (somewhat at least, as it is mentioned in the CLI help text)...